docs/shared/guides/module-federation/manage-library-versions-with-module-federation.md

```diff
@@ -0,0 +1,148 @@
+
+## How are library versions determined?
+
+**Nx** determines the version of a library by looking at a `package.json`. If the library is an npm package, the version is determined by the version declared in the workspace `package.json`. If the library is a workspace library, the version is determined by the version in the `package.json` of the project that consumes the shared library. RemoteA consumes Counter, which is a workspace library exposed and shared by RemoteB. The version of Counter is determined by the version in RemoteB's `package.json`. If the `package.json` does not exist or the library is not declared, Nx will use the version in the `package.json` of the workspace library.
+
+![How Nx determines library versions](/shared/guides/module-federation/nx-library-version.png)
+
+There are twos ways to manage library versions with **Nx**:
+
+{% tabs %}
+{% tab label="Opt out from sharing library" %}
+
+```ts {% fileName="remote/module-federation.config.ts" %}
+import { ModuleFederationConfig } from '@nx/webpack';
+
+const config: ModuleFederationConfig = {
+  name: 'remote',
+  exposes: {
+    './Module': './src/remote-entry.ts',
+  },
+  // Determine which libraries to share
+  shared: (packageName: string) {
+    // I do not want to share this package and I will load my own version
+    if(packageName === '@acme/utils') return false;
+  }
+};
+export default config;
+```
+
+This would result in the following webpack config:
+
+```js {% fileName="webpack.config.js" %}
+module.exports = {
+  plugins: [
+    new ModuleFederationPlugin({
+      // additional config
+      name: 'remote',
+      shared: {
+        react: { singleton: true, eager: true },
+        // acme/utils will not be shared
+      },
+    }),
+  ],
+};
+```
+
+{% /tab %}
+{% tab label="Opt in to sharing library versions" %}
+
+```ts {% fileName="remote/module-federation.config.ts" %}
+import { ModuleFederationConfig } from '@nx/webpack';
+
+const config: ModuleFederationConfig = {
+  name: 'remote',
+  exposes: {
+    './Module': './src/remote-entry.ts',
+  },
+  // By not declaring a shared function, all dependencies will be shared
+};
+export default config;
+```
+
+This would result in the following webpack config:
+
+```js {% fileName="webpack.config.js" %}
+module.exports = {
+  // Additional config ignored for brevity
+  plugins: [
+    new ModuleFederationPlugin({
+      // ...
+      name: 'remote',
+      shared: {
+        react: { singleton: true, eager: true, version: '18.2.0' },
+        'acme/utils': { singleton: true, eager: true, version: '1.0.0' }, // <--- This version is determined by the logic discussed earlier
+      },
+    }),
+  ],
+};
+```
+
+### Example
+
+Here is a working example of how to opt in and out of sharing library versions: https://github.com/jaysoo/module-federation-example
```

Comment:
There is a Component for showing an example repository: https://github.com/nrwl/nx/tree/master/docs#github-repositories

We should use that here

Reply:
Yea i'll update thanks.